doc/api/cli.md

```diff
@@ -121,6 +121,13 @@ added: v6.0.0
 
 Print stack traces for process warnings (including deprecations).
 
+### `--redirect-warnings=file`
```

Comment:
Sorry, missed the env var version, with that I'm only `-0`.

I was present in that conversation, and I don't think this addresses the experience of node users.

When doing npm installs, the only code that runs is npm's, so its not clear to me why we would want to silence the deprecation warnings. npm should just fix any of their deps to not cause these warnings, they are responsible citizens, they do that.

When using modules installed by npm, then I guess we don't care, usually, and can't do anything about them as a consumer of the module (other than report bugs). But, this redirect doesn't distringuish between deprecation warnings from our dependencies, and deprecations from out code itself, and node can't distinguish, it doesn't really know.

I think in the future it will look like an ad-hoc hack to deal with a more general problem, and we'll have to keep it around for a long time until we deprecate it.

That said, I am not prepared to argue more strongly against it. If noone else chimes in to object, its OK by me.